### PR TITLE
tests - simple managed test env

### PR DESCRIPTION
**Pull request by famartinrh**
Adds ability to test suite to locally deploy the registry using different storage options.
Executes the integration tests on GH actions for streams, jpa and infinispan

### CODE REVIEW

File: .github/workflows/verify.yaml [hunk 1]
```diff
@@ -35,9 +35,7 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-maven-
       - name: Build All Variants
-        run: mvn clean install -Pprod -Pjpa -Pinfinispan -Pkafka -Pstreams -Pasyncmem -DskipTests
-      - name: Verify (Standard)
-        run: mvn test package -Pprod -Pjpa -Pasyncmem
```
Comment (EricWittmann) [inline .github/workflows/verify.yaml:40]: This section runs the unit tests - I don't see this added to the other job, so I think it should stay.

File: tests/pom.xml [hunk 2]
```diff
@@ -14,6 +14,7 @@
     <name>apicurio-registry-tests</name>
 
     <properties>
+        <groups></groups>
```
Comment (EricWittmann) [inline tests/pom.xml:17]: ?

File: tests/pom.xml [hunk 3]
```diff
@@ -120,21 +125,15 @@
...
             <artifactId>slf4j-api</artifactId>
-            <version>${slf4j-api.version}</version>
-            <scope>provided</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.logging.log4j</groupId>
-            <artifactId>log4j-core</artifactId>
-            <version>${slf4j-log4j.version}</version>
-            <scope>test</scope>
+            <scope>compile</scope>
```
Comment (EricWittmann) [inline tests/pom.xml:128]: Why the change from **provided** to **compile** scope for this?

File: tests/pom.xml (hunk 2 above)
Comment (famartinrh) [reply tests/pom.xml:17]: is needed for this https://github.com/Apicurio/apicurio-registry/pull/681/files#diff-5337bf8d5cd79ec462f8b8d2c5fe2d0bR259
and the value is set when using one of the profiles I added at the end of the file. Or do you mean it will work anyways without setting the empty property first?

Comment (EricWittmann) [reply tests/pom.xml:17]: Oh I see.  Nevermind. :)

File: tests/pom.xml (hunk 3 above)
Comment (famartinrh) [reply tests/pom.xml:128]: my bad, there is no need for that

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (famartinrh) [reply .github/workflows/verify.yaml:40]: ok

File: tests/pom.xml [hunk 4]
```diff
@@ -358,5 +363,36 @@
                 <exclude.integration.tests>**/NothingToExclude.java</exclude.integration.tests>
             </properties>
         </profile>
+
+        <!-- Profiles to set test.storage system property, in case EXTERNAL_REGISTRY env var is set to false
+        this property will be used by the test suite to locally deploy the registry used the indicated storage variant
```
Comment (EricWittmann) [inline tests/pom.xml:368]: Should be "...**using** the indicated..."

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (famartinrh) [reply .github/workflows/verify.yaml:40]: unit tests are executed here https://github.com/Apicurio/apicurio-registry/pull/681/files#diff-93216a1157045809d672cb06668f4bb2R38

Comment (EricWittmann) [reply .github/workflows/verify.yaml:40]: Ah yes - you've removed the `-DskipTests` for that.  Unfortunately I think that will fail for the **streams** and **kafka** profiles still...

**Comment by famartinrh** [comment]
@carlesarnal I changed the log provider to logback

**Comment by famartinrh** [comment]
@EricWittmann how do you see this PR? I think we should consider merging this, if tests fail now it mean they are flaky and need some fixes or we should consider reduce the number of tests executed in this github test runs... anyway we should consider finish this thing and finally merge it, it's been some time now since I started working on this PR.
I consider having more complete integration testing in github actions is really important as well as having everyone contributing to integration tests to grow them and make them more stable, so this PR could be the first step in that direction.

**Comment by EricWittmann** [comment]
Hey @famartinrh I agree - I think this is good to merge.  I've just been catching up with everything now that I'm back.  But this looks good to me.
